File: frontend/BusinessFlow/src/app/modules/company/services/subscription.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from '../../../core/services/api.service';

export interface PlanInfo {
  name: string;
  description: string;
  price: number;
  durationDays: number;
  featured: boolean;
}

export interface SubscriptionResponse {
  id: number;
  plan: string;
  amount: number;
  currency: string;
  startDate: string;
  endDate: string;
  status: string;
  paymentMethod: string;
  durationMonths: number;
  autoRenew: boolean;
}

export interface CheckoutResponse {
  success: boolean;
  gatewayPageUrl: string;
  sessionKey: string;
  tranId: string;
  message: string;
}

@Injectable({ providedIn: 'root' })
export class SubscriptionService {
  private readonly endpoint = '/company/subscription';
  constructor(private api: ApiService) {}

  getPlans(): Observable<PlanInfo[]> {
    return this.api.get<PlanInfo[]>(`${this.endpoint}/plans`);
  }

  getCurrentSubscription(): Observable<SubscriptionResponse> {
    return this.api.get<SubscriptionResponse>(`${this.endpoint}/current`);
  }

  checkout(plan: string, cusName: string, cusEmail: string): Observable<CheckoutResponse> {
    return this.api.post<CheckoutResponse>(`${this.endpoint}/checkout`, { plan, cusName, cusEmail });
  }

  activate(tranId: string): Observable<{ status: string; tranId: string }> {
    return this.api.post<{ status: string; tranId: string }>(`${this.endpoint}/activate/${tranId}`, {});
  }
}

File: backend/BusinessOS/src/main/java/com/businessos/core/subscription/SubscriptionService.java
package com.businessos.core.subscription;

import com.businessos.shared.payment.sslcommerz.SslCommerzInitResponse;

import java.util.List;

public interface SubscriptionService {
    List<PlanInfo> getPlans();
    SubscriptionResponse getCurrentSubscription();
    SslCommerzInitResponse checkout(String planName, String cusName, String cusEmail);
    void activateAfterPayment(String tranId);
    void cancelSubscription(Long id);
    void expireOverdueSubscriptions();
    void sendRenewalReminders();
    void sendExpiryReminders();

    record PlanInfo(String name, String description, int price, int durationDays, boolean featured) {}
}
